src/main/java/com/answerdigital/answerking/controller/CategoryController.java

```diff
@@ -76,6 +81,18 @@ public ResponseEntity<Category> getCategoryById(@PathVariable @NotNull final Lon
         return new ResponseEntity<>(categoryService.findById(categoryId), HttpStatus.OK);
     }
 
+    @Operation(summary = "Get all products in a category.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "When all the products have been returned.",
+                content = { @Content(mediaType = "application/json", schema = @Schema(implementation = Category.class)) }),
+        @ApiResponse(responseCode = "404", description = "When the category with the given id does not exist.",
+                content = { @Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class)) })
+    })
+    @GetMapping("/{categoryId}/products")
+    public ResponseEntity<Collection<Product>> fetchProductsByCategory(@PathVariable @NotNull final Long categoryId) {
+        return new ResponseEntity<>(productService.findProductsByCategoryId(categoryId), HttpStatus.OK);
```

Comment:
Wouldn't it be better if this talked to the CategoryService and returned all products within the specified category?

Reply:
good catch, I moved it to the CategoryService.